packages/infrastructure/authentik_github_sso/main.tf

```diff
@@ -0,0 +1,150 @@
+## Upload the logo
+###########################################################################
+
+resource "random_id" "logo" {
+  prefix      = "github-"
+  byte_length = 8
+}
+
+resource "kubernetes_config_map_v1_data" "media" {
+  metadata {
+    name      = var.media_configmap
+    namespace = var.authentik_namespace
+  }
+  data = {
+    "${random_id.logo.hex}.svg" = file("${path.module}/github.svg")
+  }
+  field_manager = random_id.logo.hex
+  force         = true
+}
+
+###########################################################################
+## Cert Config
+###########################################################################
+
+// These certs are only used for their random cryptographic
+// material to sign the SAML assertions. There is no
+// need to use cert-manager to manage them,
+// especially since they need to be manually uploaded to github
+// every time they rotate
+resource "tls_private_key" "signing" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "tls_self_signed_cert" "signing" {
+  private_key_pem = tls_private_key.signing.private_key_pem
+  subject {
+    common_name  = var.authentik_domain
+    organization = var.organization_name
+  }
+  validity_period_hours = 24 * 365 * 10
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth",
+  ]
+}
+
+resource "authentik_certificate_key_pair" "signing" {
+  name             = "github-signing-certs"
+  certificate_data = tls_self_signed_cert.signing.cert_pem
+  key_data         = tls_private_key.signing.private_key_pem
+}
+
+###########################################################################
+## IdP Config
+###########################################################################
+
+data "authentik_flow" "default_authorization_flow" {
+  slug = "default-provider-authorization-implicit-consent"
+}
+
+data "authentik_property_mapping_provider_saml" "email" {
+  managed = "goauthentik.io/providers/saml/email"
+}
+
+resource "authentik_provider_saml" "github" {
+  name               = "github"
+  authorization_flow = data.authentik_flow.default_authorization_flow.id
+  property_mappings  = []
+  acs_url            = var.acs_url
+  sp_binding         = "post"
+  issuer             = local.issuer
+  audience           = var.audience
+  name_id_mapping    = data.authentik_property_mapping_provider_saml.email.id
+  signing_kp         = authentik_certificate_key_pair.signing.id
+}
+
+resource "authentik_application" "github" {
+  name              = "Github"
+  slug              = "github"
+  protocol_provider = authentik_provider_saml.github.id
+  meta_description  = var.ui_description
+  meta_publisher    = "Panfactum"
+  meta_icon         = "https://${var.authentik_domain}/media/public/${random_id.logo.hex}.svg"
+  open_in_new_tab   = true
+}
+
+data "authentik_group" "superusers" {
+  name = "superusers"
+}
+
+resource "authentik_policy_binding" "superuser_access" {
```

Comment:
Why is superuser access different?

Reply:
It seems like this would be set up twice based on `local.combined_allowed_groups`